pkg/helpers/crd.go

```diff
@@ -41,6 +42,10 @@ func CreateOrUpdateGitRepo(obj metav1.Object, gvk schema.GroupVersionKind, templ
 		},
 	}
 
+	// the git repository may not be available right after the creation. So let's give k8s a bit of time to
+	// make it available.
+	defer time.Sleep(10 * time.Millisecond)
```

Comment:
This look suspicious. Is it really necessary?

Reply:
I was able to produce "NotFound" errors quite consistently without this wait. Not sure what causes that issue though...

It only happens with Tenants, too, as between `CreateOrUpdateGitRepo()` and `GetGitRepoURLAndHostKeys()` are no other statements.

But I could also implement the `RetryOnConflict()` I used for the GitRepo reconcile.

Reply:
Using `retry.OnError()` seems to work fine for this use case.